enricher/eks/metric.go

```diff
@@ -0,0 +1,91 @@
+package eks
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/canva/amazon-kinesis-streams-for-fluent-bit/enricher/mappings"
+	"github.com/canva/amazon-kinesis-streams-for-fluent-bit/metricserver"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+type EnricherMetric struct {
+	meter metric.Meter
+
+	outputRecordCount  metric.Int64Counter
+	outputDroppedCount metric.Int64Counter
+}
+
+func WithMetricServer(ms *metricserver.MetricServer) EnricherConfiguration {
+	return func(e *Enricher) error {
+		meter := ms.GetMeter("github.com/canva/amazon-kinesis-streams-for-fluent-bit/enricher/eks")
+		outputRecordCount, err := meter.Int64Counter("fluentbit.output.record", metric.WithDescription("output record counter"))
+
+		if err != nil {
+			return err
+		}
+
+		outputDroppedCount, err := meter.Int64Counter("fluentbit.output.dropped.body", metric.WithDescription("output dropped record counter where log or message does not exist"))
+
+		if err != nil {
+			return err
+		}
+
+		e.metric = &EnricherMetric{
+			meter:              ms.GetMeter("github.com/canva/amazon-kinesis-streams-for-fluent-bit/enricher/eks"),
+			outputRecordCount:  outputRecordCount,
+			outputDroppedCount: outputDroppedCount,
+		}
+
+		return nil
+	}
+}
+
+func (e *Enricher) AddRecordCount(record map[interface{}]interface{}, logType int) {
+	if e.metric == nil {
+		return
+	}
+
+	var serviceName = inferServiceName(record, logType)
+
+	e.metric.outputRecordCount.Add(context.TODO(), 1, metric.WithAttributes(attribute.Key(mappings.RESOURCE_SERVICE_NAME).String(serviceName)))
+}
+
+func (e *Enricher) AddDropCount() {
+	if e.metric == nil {
+		return
+	}
+
+	e.metric.outputDroppedCount.Add(context.TODO(), 1)
+}
+
+func inferServiceName(record map[interface{}]interface{}, logType int) string {
+	// fallback in case unable to find service name.
+	var serviceName interface{} = "_missing"
+
+	k8sPayload, ok := record[mappings.KUBERNETES_RESOURCE_FIELD_NAME].(map[interface{}]interface{})
+	// k8s field exists, set default to container name.
+	if ok {
+		serviceName = k8sPayload[mappings.KUBERNETES_CONTAINER_NAME]
+	}
+
+	switch logType {
+	case TYPE_APPLICATION:
+
+		labels, labelsExist := k8sPayload[mappings.KUBERNETES_LABELS_FIELD_NAME].(map[interface{}]interface{})
```

Comment:
nit: Can't we just change the type cast here to `map[interface{}]string` or even `map[string]string`? So we can set `serviceName` to string and avoid the `fmt.Sprintf` at the end. Same for resources

Reply:
Also can we change this to the same way we are checking for `RESOURCE_FIELD_NAME`

Reply:
Regarding `map[interface{}]string`, I encountered an issue in deployment `telemetry-test` where it fails to cast `interface{}` to `string`.

```
panic: interface conversion: interface {} is []uint8, not string [recovered]
        panic: interface conversion: interface {} is []uint8, not string
```

This is why I have to do it as `fmt.Sprintf` but let me know if there a better way to tackle this.

Sidenote - I want to also say that this `map[interface{}]interface{}` is really painful to deal with 😢

Reply:
Oh ok so then you can probably use type `rune` and then use `string()` to convert it. I think it still would be nicer than `fmt.Sprintf` 😄

Reply:
Do you mean like this?
```
func inferServiceName(record map[interface{}]interface{}, logType LogType) string {
	// fallback in case unable to find service name.
	var serviceName []rune = []rune(mappings.PLACEHOLDER_MISSING_KUBERNETES_METADATA)

	k8sPayload, ok := record[mappings.KUBERNETES_RESOURCE_FIELD_NAME].(map[interface{}]interface{})
	// k8s field exists, set default to container name.
	if ok {
		serviceName = k8sPayload[mappings.KUBERNETES_CONTAINER_NAME].([]rune)
	}

	switch logType {
	case TYPE_APPLICATION:
		if labels, ok := k8sPayload[mappings.KUBERNETES_LABELS_FIELD_NAME].(map[interface{}][]rune); ok {
			if val, ok := labels[mappings.KUBERNETES_LABELS_NAME]; ok {
				serviceName = val
			}
		}
	case TYPE_HOST:
		if resource, ok := record[mappings.RESOURCE_FIELD_NAME].(map[interface{}][]rune); ok {
			if val, ok := resource[mappings.RESOURCE_SERVICE_NAME]; ok {
				serviceName = val
			}
		}
	}

	return string(serviceName)
}
```
I think this is a good approach but it's tricky because I believe the incoming data is in bytes which I have to convert to string anyways. With my current tests, I get a conversion error when trying strings or []byte in test.

WDYT?

Reply:
yeah it seems either way is confusing so we may as well leave as it is